lib/src/mistralai_client_dart_base.dart

```diff
@@ -19,8 +24,38 @@ class MistralAIClient {
   final int maxRetries;
   final http.Client _client;
 
+  Future<String> _getRequest<T>(String endpoint) async {
```

Comment:
Since we're not using any HTTP client libraries I think we should put all the HTTP related logic in some sort of a "service" or "helper" class 🤔 So please move it to a separate class, ideally in the "data" layer of the SDK 😉

Reply:
We agreed on a call to leave it as it is as it will be a part of https://github.com/nomtek/mistralai_client_dart/issues/13